lib/compress/zstd_compress_internal.h

```diff
@@ -1264,6 +1264,24 @@ MEM_STATIC void ZSTD_debugTable(const U32* table, U32 max)
 
 #endif
 
+/* Short Cache */
+
+typedef enum {
+    /* TODO: this is a good place to document short cache */
+    ZSTD_tfp_forCCtx = 0,
+    ZSTD_tfp_forCDict = 1
+} ZSTD_tableFillPurpose_e;
+
+#define ZSTD_SHORT_CACHE_TAG_BITS 8
+#define ZSTD_SHORT_CACHE_TAG_MASK ((1u << ZSTD_SHORT_CACHE_TAG_BITS) - 1)
```

Comment:
One note: it looks to me like there's actually no reason that the index/tag split needs to be byte-aligned. I guess maybe it means we can skip the mask operation with a `movb` instead. I'm not sure how important that is.

I'm just wondering if there's a motivation for adjusting the split point. Is that something you thought about?

Reply:
I went with 8 bits because that's what rowHash uses. There's no reason to increase the number of bits, since the collision probability is already less than 1% (so we don't pay branch mispredict tax from collisions). The only reason to reduce the number of bits would be to support dictionaries larger than 16MB, which we don't care about.

If we want to take short cache out of CDict-world and into the main hashtables, then there would be a stronger reason to reduce the number of bits. I have tried to set everything up so we can just change the macro value and everything will propagate automatically in case that does happen.

Reply:
Keeping 8-bits allows you to access the tag with a sub-register. E.g. if the index+tag is in `%rax` then the tag is `%al`. And you can `cmp %al, %bl` to compare about needing to mask.

The compiler should translate the mask + comparison into a comparison of the low byte sub-register.

Now, I don't know if that matters for performance, or if a mask would be just as fast. Just pointing out that there is a difference.